deployments/gpu-operator/reconfigure-mig.go

```diff
@@ -167,21 +177,21 @@ func reconfigure(nodeName string, migConfigFile string, selectedMIGConfig string
 			defaultGPUClientsNamespace, noRestartK8sDaemonsetsOnExit, currentLabels
 	*/
 	if err := restartingGPUClients(currentLabels, withShutdownHostGPUClients, defaultGPUClientsNamespace, hostRootMount, hostGPUClientServices, nodeName); err != nil {
-		setStateAndExit("failed", 1, currentLabels, hostGPUClientServices, withShutdownHostGPUClients, hostRootMount, nodeName)
+		setState("failed", 1, currentLabels, hostGPUClientServices, withShutdownHostGPUClients, hostRootMount, nodeName)
 		return err
 	}
 
-	setStateAndExit("success", 0, currentLabels, hostGPUClientServices, withShutdownHostGPUClients, hostRootMount, nodeName)
+	setState("success", 0, currentLabels, hostGPUClientServices, withShutdownHostGPUClients, hostRootMount, nodeName)
 	return nil
 }
 
-func setStateAndExit(state string, exitCode int, currentLabels map[string]string, hostGPUClientServices []string, withShutdownHostGPUClients bool, hostRootMount string, nodeName string) {
+func setState(state string, exitCode int, currentLabels map[string]string, hostGPUClientServices []string, withShutdownHostGPUClients bool, hostRootMount string, nodeName string) {
```

Comment:
What may make this function easier to reason about and use is to have it accept an `error` instead of an `exitCode` and also return an `error` This would mean that we would use it as: `return setState("failed", err, ...)` or `return setState("success", nil, ...)` in our calling function.

Also note that this is still running `os.Exit()` at the end which is not our desired flow.